crates/ntx-builder/src/transaction.rs

```diff
@@ -100,23 +103,23 @@ impl NtxContext {
 
                     rt.block_on(
                         async move {
-                            let mut notes = notes
+                            let notes = notes
                                 .into_iter()
-                                .map(|note| InputNote::Unauthenticated { note: note.into() })
-                                .collect::<Vec<_>>();
-                            // We shuffle the notes here to prevent having a failing note always in
-                            // front.
-                            notes.shuffle(&mut rand::rng());
-                            let notes = InputNotes::new(notes).map_err(NtxError::InputNotes)?;
+                                .map(|note| note.into_inner().into())
+                                .collect::<Vec<Note>>();
+                            let notes = InputNotes::from_unauthenticated_notes(notes)
+                                .map_err(NtxError::InputNotes)?;
```

Comment:
@bobbinth is this an ok way of going from `Note` -> `InputNote` (unauthenticated)? I don't see another way.